drivers/fpga/fpga_altera.h

```diff
@@ -0,0 +1,207 @@
+		 * is useful for matching responses to commands along with the ID
+		 */
+		uint32_t client_id : 4;
+	} header_t;
+};
+
+union config_status_version {
+	/* Version of the config status response */
+	uint32_t version;
+
+	struct {
+		/* update number bits */
+		uint32_t update_number : 8;
+		/* minor acds release number bits */
+		uint32_t minor_acds_release_number : 8;
+		/* major acds release number bits */
+		uint32_t major_acds_release_number : 8;
+		/* qspi flash index bits */
+		uint32_t qspi_flash_index : 8;
+	} response_version_member;
+};
+
+union config_status_pin_status {
+	uint32_t pin_status;
+
+	struct {
+		/* msel bits */
+		uint32_t msel : 4;
+		/* pmf data bits */
+		uint32_t pmf_data : 4;
+		/* reserve bits */
+		uint32_t reserved_bit : 22;
+		/* nconfig bits */
+		uint32_t nconfig : 1;
+		/* nconfig_status bits */
+		uint32_t nconfig_status : 1;
+	} pin_status_member;
+};
+
+/* Struct to store the fpga_config_status */
+struct fpga_config_status {
+	/* Response header */
+	union mailbox_response_header header;
+	/* Config state idle or config mode */
+	uint32_t state;
+	/* Version number */
+	union config_status_version version;
+	/* Pin status */
+	union config_status_pin_status pin_status;
+	/* Soft function status details */
+	uint32_t soft_function_status;
+	/* Location in the bitstream where the error occurred */
+	uint32_t error_location;
+	/* Data is non-zero only for certain errors. The contents are highly dependent
+	 * on which error was reported. The meaning of this data will not be made available to
+	 * customers and can only be interpreted by investigating the source code directly
+	 */
+	uint32_t error_details;
+};
+
+enum smc_cmd_code {
+	/* SMC COMMAND ID to disable all the bridges */
+	FPGA_ALL_BRIDGE_DISABLE = 0x00,
+	/* SMC COMMAND ID to enable all the bridges */
+	FPGA_ALL_BRIDGE_ENABLE = 0x01,
+	/* SMC Cancel Command */
+	FPGA_CANCEL = 0x03,
+	/* SMC COMMAND ID to check Reconfig status to SDM via mailbox */
+	FPGA_CONFIG_STATUS = 0x04,
+	/* SMC COMMAND ID to check Reconfig status to SDM via mailbox */
+	FPGA_RECONFIG_STATUS = 0x09
+};
+
+enum mbox_reconfig_status_resp {
+	/* Mailbox reconfig status header */
+	MBOX_RECONFIG_STATUS_HEADER,
+	/* Mailbox reconfig status state */
+	MBOX_RECONFIG_STATUS_STATE,
+	/* Mailbox reconfig status version */
+	MBOX_RECONFIG_STATUS_VERSION,
+	/* Mailbox reconfig status pin status */
+	MBOX_RECONFIG_STATUS_PIN_STATUS,
+	/* Mailbox reconfig status soft function */
+	MBOX_RECONFIG_STATUS_SOFT_FUNCTION,
+	/* Mailbox reconfig status error location */
+	MBOX_RECONFIG_STATUS_ERROR_LOCATION,
+	/* Mailbox reconfig status error details */
+	MBOX_RECONFIG_STATUS_ERROR_DETAILS
+};
+
+enum smc_request {
+	/* SMC request parameter a2 index*/
+	SMC_REQUEST_A2_INDEX = 0x00,
+	/* SMC request parameter a3 index */
+	SMC_REQUEST_A3_INDEX = 0x01
+};
+
+/* SIP SVC response private data */
+struct private_data_t {
```

Comment:
This is not typedef'ed so please remove the `_t`
```suggestion
struct private_data {
```

Reply:
done